python/cuml/test/test_thirdparty.py

```diff
@@ -0,0 +1,166 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pytest
+
+import numpy as np
+import cupy as cp
+from cuml._thirdparty.sklearn.utils.validation import check_X_y
+
+from cuml._thirdparty.sklearn.utils.extmath import row_norms as cu_row_norms, \
+    _incremental_mean_and_var as cu_incremental_mean_and_var
+from sklearn.utils.extmath import row_norms as sk_row_norms, \
+    _incremental_mean_and_var as sk_incremental_mean_and_var
+
+from cuml._thirdparty.sklearn.utils.sparsefuncs import \
+    inplace_csr_column_scale as cu_inplace_csr_column_scale, \
+    inplace_csr_row_scale as cu_inplace_csr_row_scale, \
+    inplace_column_scale as cu_inplace_column_scale, \
+    mean_variance_axis as cu_mean_variance_axis, \
+    min_max_axis as cu_min_max_axis
+from sklearn.utils.sparsefuncs import \
+    inplace_csr_column_scale as sk_inplace_csr_column_scale, \
+    inplace_csr_row_scale as sk_inplace_csr_row_scale, \
+    inplace_column_scale as sk_inplace_column_scale, \
+    mean_variance_axis as sk_mean_variance_axis, \
+    min_max_axis as sk_min_max_axis
+
+from cuml.test.test_preproc_utils import assert_allclose
+
+
+@pytest.fixture(scope="session")
+def random_dataset(request):
+    X = cp.random.rand(100, 10)
```

Comment:
Should these all have a fixed random state? We could create a cupy random state as a fixture and reuse that for all random creation. I'm less worried about this one fixture and more about all the other cp.random uses below

Reply:
Do you think that a `random_state` might be necessary? How would it work? Maybe a fixture could randomly output a seed out of a set of certified seeds. Other options? However I believe that a fixture cannot take another fixture as input. Because of this, the main dataset wouldn't be able to be generated from it.